internal/keepers/ocr.go

```diff
@@ -57,7 +65,31 @@ func (k *keepers) Observation(ctx context.Context, rt types.ReportTimestamp, _ t
 	// should be more uniform for all nodes
 	keys := keyList(filterUpkeeps(results, ktypes.Eligible))
 
-	b, err := limitedLengthEncode(keys, maxObservationLength)
+	// Check limit
+	if len(keys) > observationKeysLimit {
+		keys = keys[:observationKeysLimit]
+	}
+
+	latestBlock, err := k.service.LatestBlock(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("%w: failed to get latest block", err)
+	}
+
+	obs := &ktypes.UpkeepObservation{
+		BlockKey:          ktypes.BlockKey(latestBlock.String()),
+		UpkeepIdentifiers: []ktypes.UpkeepIdentifier{},
+	}
+
+	if len(keys) > 0 {
+		var identifiers []ktypes.UpkeepIdentifier
+		for _, upkeepKey := range keys {
+			_, upkeepID, _ := upkeepKey.BlockKeyAndUpkeepID()
+			identifiers = append(identifiers, upkeepID)
```

Comment:
not getting the blockKey from the key? 
We'll need to ensure the latestBlock we send matches the upkeep keys

Reply:
discussed offline. will get the latest block from sampling process